app/src/main/java/com/example/simbirsoft_android_practice/core/CategoryRepository.kt

```diff
@@ -13,64 +13,68 @@ import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.emitAll
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.map
+import javax.inject.Inject
 
 private const val TAG_CATEGORY_REPOSITORY = "CategoryRepository"
 private const val CATEGORIES_JSON_FILE = "categories.json"
 private const val TIMEOUT_IN_MILLIS = 5_000L
 
-class CategoryRepository(
-    private val extractor: JsonAssetExtractor,
-    private val categoryDao: CategoryDao,
-) {
-    private val gson = Gson()
-    private var isDataLoaded = false
+class CategoryRepository
+    @Inject
+    constructor(
```

Comment:
Обычно это делают в одну строчку, хотя для тебя это может быть немного непривычно:
```suggestion
class CategoryRepository @Inject constructor(
```